client/matching/loadbalancer.go

```diff
@@ -134,3 +142,34 @@ func (lb *defaultLoadBalancer) pickPartition(
 
 	return fmt.Sprintf("%v%v/%v", taskListPartitionPrefix, taskList.GetName(), p)
 }
+
+func (lb *defaultLoadBalancer) GetAllPartitions(
+	domainID string,
+	taskList shared.TaskList,
+	taskListType int,
+) ([]string, error) {
+	domainName, err := lb.domainIDToName(domainID)
+	var partitionKeys []string
+	if err != nil {
+		return partitionKeys, err
+	}
+
+	taskListName := taskList.GetName()
+	if strings.HasPrefix(taskListName, taskListPartitionPrefix) {
+		// get the original task list name if the name contains partition prefix
+		taskListName = strings.Split(taskListName, "/")[1]
+	}
+
+	partitionKeys = append(partitionKeys, taskListName)
+
+	n := lb.nWritePartitions(domainName, taskListName, taskListType)
+	if n <= 0 {
+		return partitionKeys, nil
+	}
+
+	for i := 1; i < n; i++ {
```

Comment:
Off by one error?

Reply:
It is off by one because there is no prefix for the first partition.